### PR TITLE
add admin operation logs page

### PR DESCRIPTION
**Pull request by imwhatiam**

### CODE REVIEW

File: seahub/templates/js/sysadmin-templates.html [hunk 1]
```diff
@@ -839,3 +843,75 @@ <h2 class="alc">{% trans "No members" %}</h2>
...
+    <% if (app.pageOptions.is_pro && app.pageOptions.enable_sys_admin_view_repo) { %>
+    <td>{% trans "Transfer library" %} <a href="#libs/<%- detail['id'] %>/"><%- detail['name'] %></a> {% trans "from" %} <a href="{{ SITE_ROOT }}useradmin/info/<%- detail['from'] %>/"><%- detail['from'] %></a> {% trans "to" %} <a href="{{ SITE_ROOT }}useradmin/info/<%- detail['to'] %>/"><%- detail['to'] %></a></td>
+    <% } else { %>
+    <td>{% trans "Transfer library" %} <%- detail['name'] %> {% trans "from" %} <a href="{{ SITE_ROOT }}useradmin/info/<%- detail['from'] %>/"><%- detail['from'] %></a> {% trans "to" %} <a href="{{ SITE_ROOT }}useradmin/info/<%- detail['to'] %>/"><%- detail['to'] %></a></td>
+    <% } %>
+    <% } %>
+
+    <% if (operation=='group_create') { %>
+    <td>{% trans "Create Group" %}</td>
+    <td>{% trans "Create group" %} <span><a href="#groups/<%- detail['id'] %>/"><%- detail['name'] %></span></td>
```
Comment (llj) [inline seahub/templates/js/sysadmin-templates.html]: m

File: seahub/admin_log/models.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            filtered_logs = logs.filter(operation=operation)
+        else:
+            filtered_logs = logs
+
+        return filtered_logs
+
+class AdminLog(models.Model):
+    email = models.EmailField(db_index=True)
+    operation = models.CharField(max_length=255, db_index=True)
+    detail = models.CharField(max_length=255)
```
Comment (xiez) [inline seahub/admin_log/models.py]: 改成 2048

File: seahub/api2/endpoints/admin/libraries.py [hunk 3]
```diff
@@ -150,24 +151,33 @@ def delete(self, request, repo_id, format=None):
...
 
         try:
             seafile_api.remove_repo(repo_id)
-            repo_deleted.send(sender=None,
-                              org_id=-1,
-                              usernames=related_usernames,
-                              repo_owner=repo_owner,
-                              repo_id=repo_id,
-                              repo_name=repo.name)
+            related_usernames = seaserv.get_related_users_by_repo(repo_id)
```
Comment (xiez) [inline seahub/api2/endpoints/admin/libraries.py]: 如果repo损坏？

File: seahub/urls.py [hunk 4]
```diff
@@ -212,6 +213,8 @@
     url(r'^api/v2.1/admin/trash-libraries/$', AdminTrashLibraries.as_view(), name='api-v2.1-admin-trash-libraries'),
     url(r'^api/v2.1/admin/trash-libraries/(?P<repo_id>[-0-9a-f]{36})/$', AdminTrashLibrary.as_view(), name='api-v2.1-admin-trash-library'),
     url(r'^api/v2.1/admin/shares/$', AdminShares.as_view(), name='api-v2.1-admin-shares'),
+    # TODO
+    url(r'^api/v2.1/admin/admin-logs/$', AdminLogs.as_view(), name='api-v2.1-admin-admin-logs'),
```
Comment (xiez) [inline seahub/urls.py]: todo

File: seahub/admin_log/models.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+        model.save()
+
+        return model
+
+    def get_admin_logs(self, email=None, operation=None):
+
+        logs = super(AdminLogManager, self).all()
+
+        if email and operation:
```
Comment (xiez) [inline seahub/admin_log/models.py]: if email is not None: logs = logs.filter(..)
if operation is not None : logs = ...

File: seahub/api2/endpoints/admin/libraries.py [hunk 6]
```diff
@@ -150,24 +151,33 @@ def delete(self, request, repo_id, format=None):
...
             return api_error(status.HTTP_500_INTERNAL_SERVER_ERROR, error_msg)
 
+        # send admin operation log signal
+        admin_op_detail = {
+            "id": repo_id,
+            "name": repo_name,
+            "owner": repo_owner,
+        }
+        admin_operation.send(sender=None, admin_name=request.user.username,
+                operation='repo_delete', detail=admin_op_detail)
```
Comment (xiez) [inline seahub/api2/endpoints/admin/libraries.py]: 字符串硬编码

**Pull request merged**